include/inexor/vulkan-renderer/imgui/imgui.hpp

```diff
@@ -0,0 +1,101 @@
+#include "inexor/vulkan-renderer/wrapper/shader.hpp"
+#include "inexor/vulkan-renderer/wrapper/swapchain.hpp"
+#include "inexor/vulkan-renderer/wrapper/texture.hpp"
+
+#include <glm/glm.hpp>
+#include <imgui.h>
+#include <spdlog/spdlog.h>
+#include <vma/vk_mem_alloc.h>
+#include <vulkan/vulkan.h>
+
+#include <assert.h>
+#include <iomanip>
+#include <memory>
+#include <sstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <vector>
+
+namespace inexor::vulkan_renderer::imgui {
+class ImGUIOverlay {
+    const wrapper::Device &m_device;
+    const wrapper::Swapchain &m_swapchain;
+    const VkPhysicalDevice m_graphics_card;
+    const VmaAllocator m_vma_allocator;
+    const VkQueue m_data_transfer_queue;
+    const std::uint32_t m_data_transfer_queue_family_index;
+
+    VkPipelineLayout m_pipeline_layout;
+    VkSampleCountFlagBits rasterizationSamples{VK_SAMPLE_COUNT_1_BIT};
+
+    bool visible{true};
+    float m_scale{1.0f};
+
+    std::unique_ptr<wrapper::MeshBuffer> m_imgui_mesh;
+    std::unique_ptr<wrapper::Texture> m_imgui_texture;
+    std::unique_ptr<wrapper::RenderPass> m_renderpass;
+    std::unique_ptr<wrapper::Shader> m_vert_shader;
+    std::unique_ptr<wrapper::Shader> m_frag_shader;
+    std::unique_ptr<wrapper::CommandPool> m_command_pool;
+    std::unique_ptr<wrapper::ResourceDescriptor> m_descriptor;
+    std::unique_ptr<wrapper::GraphicsPipeline> m_pipeline;
+
+    std::uint32_t subpass{0};
+    std::uint32_t vertexCount{0};
+    std::uint32_t indexCount{0};
+
+    std::vector<VkPipelineShaderStageCreateInfo> shaders;
+    std::vector<std::unique_ptr<wrapper::CommandBuffer>> m_command_buffers;
+    std::vector<std::unique_ptr<wrapper::Framebuffer>> m_framebuffers;
+
+    // TODO() Implement an RAII wrapper for push constants!
+    struct PushConstBlock {
+        glm::vec2 scale;
+        glm::vec2 translate;
+    } pushConstBlock;
+
+public:
+    ImGUIOverlay(const ImGUIOverlay &) = delete;
+    ImGUIOverlay(ImGUIOverlay &&) noexcept;
+
+    ImGUIOverlay &operator=(const ImGUIOverlay &) = delete;
+    ImGUIOverlay &operator=(ImGUIOverlay &&) noexcept;
+
+    /// @brief Default constructor
+    /// @param device [in] A reference to the device wrapper.
+    /// @param swapchain [in] A reference to the swapchain.
+    /// @param graphics_card [in] The graphics card handle.
+    /// @param vma_allocator [in] The Vulkan memory allocator handle
+    /// @param data_transfer_queue [in] The Vulkan transfer queue handle.
+    /// @param data_transfer_queue_family_index [in] The queue family index for data transfer.
+    ImGUIOverlay(const wrapper::Device &device, const wrapper::Swapchain &swapchain,
+                 const VkPhysicalDevice graphics_card, const VmaAllocator vma_allocator,
+                 const VkQueue data_transfer_queue, const std::uint32_t data_transfer_queue_family_index);
+
+    ~ImGUIOverlay();
+
+    bool updated = false;
```

Comment:
Public not thread safe value. Also

```suggestion
    bool m_updated = false;
```

Reply:
Done.